### ISSUE TITLE
Hitting a couple segfaults when running node2vec embedding algorithm

### ISSUE THREAD

**Issue by abhayan7** [networkit/networkit#1011]
Env:
- NetworKit v10.0
- Python 3.7.12

While experimenting with generating node2vec embeddings through NetworKit, I came across a couple scenarios throwing segmentation faults:

1. When I try to generate an embedding on any directed graph, I seem to hit a segfault consistently. Does not occur with undirected graphs (at least, in this scenario).

Sample code:
```python
import networkit as nk

G = nk.graph(n=5, directed=True)
G.addEdge(1, 3)
G.addEdge(2, 4)
G.addEdge(4, 0)

node2vec = nk.embedding.Node2Vec(G)

## node2vec.run() hits segfault:
node2vec.run()
```

2. With undirected graphs, I hit a segfault intermittently if I have an isolated node with no edges. I ran the following code 20 times, and hit the segfault in 12 of the 20 runs. Does not occur if I add an edge to node 0.

Sample code:
```python
import networkit as nk

nk.setNumberOfThreads(1)

G = nk.graph(n=5, directed=False)
G.addEdge(1, 3)
G.addEdge(2, 4)
## No edge on node 0 in this scenario

node2vec = nk.embedding.Node2Vec(G)

## node2vec.run() hits segfault intermittently:
node2vec.run()
```

**Comment by fabratu** [comment]
Hi, thx for posting the issue.

We could reproduce the segfaults and are currently investigating.

**Comment by fabratu** [comment]
Closing this issue, as the posted problems should be fixed with release 10.1. 

If there is something else or the update is not working properly, feel free to reopen the issue.
